RecoLocalCalo/HGCalRecAlgos/src/RecHitTools.cc

```diff
@@ -64,21 +82,30 @@ void RecHitTools::getEventSetup(const edm::EventSetup& es) {
   es.get<CaloGeometryRecord>().get(geom);
 
   geom_ = geom.product();
-  auto geomEE = dynamic_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE));
-  fhOffset_ = (geomEE->topology().dddConstants()).layers(true);
-  unsigned int wmaxEE = 1 + (geomEE->topology().dddConstants()).waferMax();
-  auto geomFH = dynamic_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCHEF));
-  bhOffset_ = fhOffset_ + (geomFH->topology().dddConstants()).layers(true);
-  unsigned int wmaxFH = 1 + (geomFH->topology().dddConstants()).waferMax();
+  unsigned int wmaxEE(0), wmaxFH(0);
+  if (mode_ == 0) {
+    auto geomEE = dynamic_cast<const HGCalGeometry*>(geom_->getSubdetectorGeometry(DetId::Forward,ForwardSubdetector::HGCEE));
```

Comment:
get it from an event setup?

Reply:
RecHitTools is Marco's code. I just added what changes to be made for the new geometry. I do not want to change the design of this. Later Marco can take care of the changes in the RecHitTools code

Reply:
@rovere FYI